gym-unity/gym_unity/envs/__init__.py

```diff
@@ -40,7 +40,7 @@ def __init__(
         """
         Environment initialization
         :param unity_env: The Unity BaseEnv to be wrapped in the gym. Will be closed when the UnityToGymWrapper closes.
-        :param use_visual: Whether to use visual observation or vector observation.
+        :param use_visual: Whether to use visual observations.
```

Comment:
I think this is breaking the assumption that if use_visual is enabled, the observation will be a np.array and not a list of np.array.
I think this change is more suited for `allow_multiple_visual_obs`.
Maybe renaming `allow_multiple_visual_obs` to `allow_multiple_obs` and always return a list of (all visual and vector) observations would be better?

Reply:
From my point of view your suggestion would be the better and cleaner solution. I tried to keep it as compatible as possible, but the changes one has to make to get his or her code working again would be rather simple. I'll add this to the PR.

Reply:
Just realizing: What exactly should `use_visual` mean then? Does it mean there are one or more visual observations? That's would the name implies. However, that does not mean that the observation will be a numpy array. 

I think it would make more sense to have three parameters here: 
```
use_vector_only: Use only vector observations. Environments returns an np.array
use_single_visual: Use a single visual observation. Environments returns an np.array
allow_multiple_obs: Allows vector and multiple visual observations. Environment returns a list of observations with the last element containing the array of vector observations. 

```
Let me know what you think.

Reply:
That looks pretty nice. I wonder if an **enum** would be more appropriate in that case, a user could do `use_vector_only=True, use_visual_only=True` by mistake.

I think the simplest would be a single boolean : `use_multiple_obs` (default false)
 - If true, returns a list of np_array as observations
 - If false, will return a single np_array **AND** will return the first visual observation if any (over vector observation)

What do you think ?

Reply:
That's probably even better. That way you know whether you get a list or an array. I can take care of that tomorrow and commit it here.

Reply:
That would be awesome, thank you for your contributions !